view/scene.py

```diff
@@ -54,4 +63,77 @@ def plot_quivers(self):
 		
 		plt.axis([-10, 10, -10, 10])
 		self.fig.canvas.draw()
+
+class ContinuousPlotter(object):
+	def __init__(self, get_next_frames_callback):
+		self.get_next_frames_callback = get_next_frames_callback
+
+		self.dt = 0.5
+		self.frame_batch_duration = 10 * self.dt
+		self.current_time = 0.0
+		self.last_update_time = None
+		self.update_period = self.frame_batch_duration
+
+		assert(self.update_period <= self.frame_batch_duration)
+
+		self.max_queue_size = 10
+
+		self.frame_buffer_queue = queue.Queue( maxsize = self.max_queue_size )
```

Comment:
Ohhhhhhh.... `queue.Queue` IS threadsafe. Didn't know it 😮 . Thanks for teaching me this.

Reply:
Yeah, I use this for threadsafety but if I don't want threadsafety (the overheard etc.) I use `collections.deque`

This queue, is pretty convenient, the get() and put() methods are threadsafe and get() blocks until the queue is empty, while if the queue has a maxsize then the put() blocks until the size drops below maxsize.